File: samples/javascript_nodejs/54.teams-task-module/models/taskmoduleresponsefactory.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

class TaskModuleResponseFactory {
    static createResponse(taskModuleInfoOrString) {
        if (typeof taskModuleInfoOrString == 'string') {
            return {
                task: {
                    type: 'message',
                    value: taskModuleInfoOrString
                }
            };
        }

        return {
            task: {
                type: 'continue',
                value: taskModuleInfoOrString
            }
        };
    }

    static toTaskModuleResponse(taskInfo) {
        return TaskModuleResponseFactory.createResponse(taskInfo);
    }
}

module.exports.TaskModuleResponseFactory = TaskModuleResponseFactory;

File: archive/samples/python/54.teams-task-module/models/task_module_response_factory.py
# Copyright (c) Microsoft Corp. All rights reserved.
# Licensed under the MIT License.

from typing import Union

from botbuilder.schema.teams import (
    TaskModuleResponse,
    TaskModuleMessageResponse,
    TaskModuleTaskInfo,
    TaskModuleContinueResponse,
)


class TaskModuleResponseFactory:
    @staticmethod
    def create_response(value: Union[str, TaskModuleTaskInfo]) -> TaskModuleResponse:
        if isinstance(value, TaskModuleTaskInfo):
            return TaskModuleResponse(task=TaskModuleContinueResponse(value=value))
        return TaskModuleResponse(task=TaskModuleMessageResponse(value=value))

    @staticmethod
    def to_task_module_response(task_info: TaskModuleTaskInfo):
        return TaskModuleResponseFactory.create_response(task_info)
